Don't toposort nodes in non-user-facing operations (#3146)
kedro-org/kedro: RELEASE.md

## Major features and improvements
* Improved error message when passing wrong value to node.

> * Improved performance of pipeline operations by using non-toposorted nodes in internal code.

Reviewer: ```suggestion
* Improved performance of pipeline operations by avoiding unnecessary copies of the nodes.
```
